engine/crates/runtime/src/fetch.rs

```diff
@@ -20,53 +21,135 @@ impl FetchError {
 
 pub type FetchResult<T> = Result<T, FetchError>;
 
-// very minimal for now, but will be expanded as we need it.
-pub struct FetchRequest<'a> {
-    pub url: &'a url::Url,
+/// reqwest uses Url instead of Uri, so as long as it's the actual implementation underneath it's a
+/// bit of a waste to use http::Request
```

Comment:
I don't understand this comment — what's URI in this context?

Reply:
meant this: https://docs.rs/http/latest/http/uri/struct.Uri.html
instead of `url::Url`

Reply:
Aah ok, makes sense